src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const stateHistory = [];
+
+  actions.forEach((action) => {
+    let nextState = { ...currentState };
```

Comment:
[CODE STYLE] - You are creating a new copy of the `currentState` object on each iteration of the loop. This is unnecessary and can be avoided by only copying the `currentState` object when you need to modify it.